src/components/ThemeProvider/tests/ThemeProvider.test.tsx

```diff
@@ -164,4 +164,145 @@ describe('<ThemeProvider />', () => {
       return null;
     }
   });
+
+  // since we mount each `ThemeProvider` withApp, each mounted `ThemeProvider` is a nested one
+  // we may need some sort of escape hatch to rendering a theme provider by default
+  it.todo('does set overrides');
```

Comment:
This is an important piece to call out, as the changes to `ThemeProvider` that allow it to be both a parent and a child, mean that any `ThemeProvider` mounted `withApp` will be a child theme provider (`withApp` also means `withThemeProvider`), so it won't perform any default behavior.

This may be a good reason not to make `ThemeProvider` perform that dual role, as it becomes hard to test. Or it could be a good reason for providing an escape hatch to allow `AppProvider` not to render a `ThemeProvider` by default.

Reply:
Taking a quick look at this test file I can't see any obvious reason why it can't use boring old `mount` instead of `mountWithAppProvider` to sidestep the "`mountWithAppProvider` provides a `ThemeContext` but we don't want that" issue.